[NavigationDrawer] Add API to present navigation drawer

codeman7 opened the pull request:
### Context
Clients may need to animate their navigation drawer to full screen if a user interacts with it in a specific way, for example they tap a button and the navigation drawer presents in full screen. Within exploring that problem, I realized they may also want to be able to collapse the drawer if it is in full screen. For example, a user taps a button and it expands but then the user realizes they didn't want to tap that button so they want to minimize the navigation drawer back down to a smaller size. We may want to unify the API.
### Other possible solutions
1. We alternatively could use 
`UIView.animateWithDuration...` to animate the view with custom timing function if we feel that is required. Within the animation block we would still be calling `self.scrollview.contentOffset = desiredContentOffset`

2. A unified API for expanding and collapsing the navigation drawer, possibly `animateToFullScreen:(BOOL)fullScreen withCompletion:...`
This would allow clients to call `animateToFullScreen:YES` if they want to present full screen or `animateToFullScreen:NO` if they want to animate the drawer back down to its initial height. 

### The problem
We need to allow clients to animate the navigation drawer to full screen
### The fix
Add an API to allow clients to animate the navigation drawer to full screen
### Why WIP
No test have been added, we may want to have a unified API for collapsing and expanding, also missing docs.

### Related bug
#5600 

### Videos
_Screenshots are as of ~10:15am 11/14 any commits after may change them_


|  iPhone X  |  iPhone 6s  |
| --- | --- |
|![after](https://user-images.githubusercontent.com/7131294/48492104-0add1700-e7f7-11e8-9e27-0c76fc040c2e.gif)|![6s](https://user-images.githubusercontent.com/7131294/48492110-0e709e00-e7f7-11e8-9493-ea4b7abc2dfc.gif)|

codeman7 committed:
Initial commit

codeman7 committed:
Working for different screen sizes

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +447,16 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
 
+- (void)expandToFullScreen {
+  CGFloat topSafeArea = 0;
+  if (@available (iOS 11.0, *)) {
```
clang-format suggested the following change:


```
-  if (@available (iOS 11.0, *)) {
+  if (@available(iOS 11.0, *)) {
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.h:
```diff
@@ -91,6 +91,8 @@
  */
 - (CGFloat)topCornersRadiusForDrawerState:(MDCBottomDrawerState)drawerState;
 
+- (void)expandToFullScreenWithCompletion:(void (^ __nullable)(BOOL finished))completion;
```
clang-format suggested the following change:


```
-- (void)expandToFullScreenWithCompletion:(void (^ __nullable)(BOOL finished))completion;
+- (void)expandToFullScreenWithCompletion:(void (^__nullable)(BOOL finished))completion;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerViewController.h -->

material-automation commented:
## API diff detected the following changes

### NavigationDrawer

#### MDCBottomDrawerViewController

*new* method: `-animateToPreferredContentHeight:withDuration:completion:` in `MDCBottomDrawerViewController`

#### MDCBottomDrawerPresentationController

*new* method: `-animateToPreferredContentHeight:withDuration:completion:` in `MDCBottomDrawerPresentationController`

<!-- identifier: apidiff -->

codeman7 committed:
Clang

rami-a commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.m:
```diff
@@ -205,4 +205,13 @@ - (void)contentDrawerTopInset:(CGFloat)transitionToTop {
+        (MDCBottomDrawerPresentationController *)self.presentationController;
+    [bottomDrawerPresentationController expandToFullScreen];
+  }
+  completion(YES);
```
I may be wrong about this but wouldn't we want the completion block to fire after the animation is finished. I'm not sure if that would be possible with this implementation but something to consider.

codeman7 replied:
Yeah, I considered that, I don't know really where the completion block should go, if it's after the animation then it animates and "jumps" to the completed state, if it's as it stands it "jumps" to the completion as it animates, I think we want to have them animate together.

codeman7 committed:
Initial drawer explorations

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +457,18 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+  }
+  CGFloat contentYOffset = self.contentHeaderTopInset - topSafeArea;
+  CGPoint contentOffset = CGPointMake(self.scrollView.contentOffset.x, contentYOffset);
+  //self.initialDrawerFactor = 1;
```
clang-format suggested the following change:


```
-  //self.initialDrawerFactor = 1;
+  // self.initialDrawerFactor = 1;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

codeman7 committed:
Revert

codeman7 committed:
Working for small content

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +447,22 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+  [self cacheLayoutCalculations];
+  [UIView animateWithDuration:duration animations:^{
+    [self.scrollView setContentOffset:contentOffset];
+  }];
```
clang-format suggested the following change:


```
-  [UIView animateWithDuration:duration animations:^{
-    [self.scrollView setContentOffset:contentOffset];
-  }];
+  [UIView animateWithDuration:duration
+                   animations:^{
+                     [self.scrollView setContentOffset:contentOffset];
+                   }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

codeman7 committed:
Address header

codeman7 committed:
Changes before initial drawer

codeman7 committed:
Content not working

codeman7 committed:
Animates correctly

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,45 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    topSafeArea = self.view.safeAreaInsets.top;
+  }
+  CGFloat totalHeight = self.headerViewController.preferredContentSize.height +
+  self.contentViewController.preferredContentSize.height;
```
clang-format suggested the following change:


```
-  self.contentViewController.preferredContentSize.height;
+                        self.contentViewController.preferredContentSize.height;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,45 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      self->_contentHeightSurplus = NSNotFound;
+      self->_addedContentHeight = NSNotFound;
+      [self.view setNeedsLayout];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      self.initialDrawerFactor = precentageOfFullScreen;
-      self->_contentHeaderTopInset = NSNotFound;
-      self->_contentHeightSurplus = NSNotFound;
-      self->_addedContentHeight = NSNotFound;
-      [self.view setNeedsLayout];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          self.initialDrawerFactor = precentageOfFullScreen;
+          self->_contentHeaderTopInset = NSNotFound;
+          self->_contentHeightSurplus = NSNotFound;
+          self->_addedContentHeight = NSNotFound;
+          [self.view setNeedsLayout];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,45 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      self->_contentHeightSurplus = NSNotFound;
+      self->_addedContentHeight = NSNotFound;
+      [self.view setNeedsLayout];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      self.initialDrawerFactor = precentageOfFullScreen;
-      self->_contentHeaderTopInset = NSNotFound;
-      self->_contentHeightSurplus = NSNotFound;
-      self->_addedContentHeight = NSNotFound;
-      [self.view setNeedsLayout];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          self.initialDrawerFactor = precentageOfFullScreen;
+          self->_contentHeaderTopInset = NSNotFound;
+          self->_contentHeightSurplus = NSNotFound;
+          self->_addedContentHeight = NSNotFound;
+          [self.view setNeedsLayout];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,45 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    }];
+  }
+
+
```
clang-format suggested the following change:


```
-
-
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

codeman7 committed:
Fix full screen

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,44 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    topSafeArea = self.view.safeAreaInsets.top;
+  }
+  CGFloat totalHeight = self.headerViewController.preferredContentSize.height +
+      self.contentViewController.preferredContentSize.height;
```
clang-format suggested the following change:


```
-      self.contentViewController.preferredContentSize.height;
+                        self.contentViewController.preferredContentSize.height;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,44 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      [self.scrollView setContentOffset:contentOffset];
+    } completion:^(BOOL finished) {
+      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +469,44 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    } completion:^(BOOL finished) {
+      [self.scrollView setContentOffset:CGPointZero];
+      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      [self.scrollView setContentOffset:CGPointZero];
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          [self.scrollView setContentOffset:CGPointZero];
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

codeman7 committed:
Test

codeman7 committed:
Update docs

material-automation commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,39 @@ - (void)testBottomDrawerTopInset {
+  self.fakeBottomDrawer.contentViewController.preferredContentSize = CGSizeMake(200, 5000);
+  CGSize fakePreferredContentSize = CGSizeMake(200, 400);
+  MDCNavigationDrawerFakeHeaderViewController *fakeHeader =
+  [[MDCNavigationDrawerFakeHeaderViewController alloc] init];
```
clang-format suggested the following change:


```
-  [[MDCNavigationDrawerFakeHeaderViewController alloc] init];
+      [[MDCNavigationDrawerFakeHeaderViewController alloc] init];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,48 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    } completion:^(BOOL finished) {
+      [self setDrawerState:MDCBottomDrawerStateExpanded];
+      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      [self setDrawerState:MDCBottomDrawerStateExpanded];
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          [self setDrawerState:MDCBottomDrawerStateExpanded];
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,48 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      [self.scrollView setContentOffset:CGPointZero];
+      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+      [self setDrawerState:MDCBottomDrawerStateFullScreen];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      // Reset the content offset to 0 because we are going to recachelayout and reset initial
-      // drawer factor.
-      [self.scrollView setContentOffset:CGPointZero];
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-      [self setDrawerState:MDCBottomDrawerStateFullScreen];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          // Reset the content offset to 0 because we are going to recachelayout and reset initial
+          // drawer factor.
+          [self.scrollView setContentOffset:CGPointZero];
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+          [self setDrawerState:MDCBottomDrawerStateFullScreen];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

codeman7 committed:
Update

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,46 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      // drawer factor.
+      [self.scrollView setContentOffset:CGPointZero];
+      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+    }];
```
clang-format suggested the following change:


```
-    [UIView animateWithDuration:duration animations:^{
-      [self.scrollView setContentOffset:contentOffset];
-    } completion:^(BOOL finished) {
-      // Reset the content offset to 0 because we are going to recachelayout and reset initial
-      // drawer factor.
-      [self.scrollView setContentOffset:CGPointZero];
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-    }];
+    [UIView animateWithDuration:duration
+        animations:^{
+          [self.scrollView setContentOffset:contentOffset];
+        }
+        completion:^(BOOL finished) {
+          // Reset the content offset to 0 because we are going to recachelayout and reset initial
+          // drawer factor.
+          [self.scrollView setContentOffset:CGPointZero];
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+        }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

yarneo commented on components/NavigationDrawer/examples/supplemental/BottomDrawerSupplemental.swift:
```diff
@@ -68,7 +58,14 @@ class DrawerHeaderViewController: UIViewController,MDCBottomDrawerHeader {
 
   override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
     titleLabel.sizeToFit()
```
new line after super

yarneo commented on components/NavigationDrawer/examples/supplemental/BottomDrawerSupplemental.swift:
```diff
@@ -68,7 +58,14 @@ class DrawerHeaderViewController: UIViewController,MDCBottomDrawerHeader {
   override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
     titleLabel.sizeToFit()
+
```
redundant new line

yarneo commented on components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:
```diff
@@ -87,4 +87,12 @@
 
+/**
+ Expands the navigation drawer to maximum height for the content. If content is larger than the
+ screen bounds then animates to full screen.
```
Doesn't it always just go to the full height of the window? is there a case where it reaches the maximum height of the content and it doesn't expand to full screen?

codeman7 replied:
Yes, if for example the content and header are a total height of 650 and the window is 800 then it will expand to ~81% of the screen.

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -285,18 +290,7 @@ - (BOOL)isMobileLandscape {
- */
-- (CGFloat)initialDrawerFactor {
-  return [self shouldPresentFullScreen] ? 1 : (CGFloat)0.5;
+  return [self isAccessibilityMode] || [self isMobileLandscape] || _initialDrawerFactor >= 1;
```
i think there is probably a better comparison than _initialDrawerFactor >=1. Reason being is that if the initial drawer factor is close to 1, i think the drawer still goes to full screen because it adjusts to full screen if its in the middle of transition the header to cover the safe area and the status bar.

please open a follow up bug to research this is a follow up PR.

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -325,6 +319,30 @@ - (void)setDrawerState:(MDCBottomDrawerState)drawerState {
+  _initialDrawerFactor = initialDrawerFactor;
+  _contentHeaderTopInset = NSNotFound;
+  _contentHeightSurplus = NSNotFound;
+  _addedContentHeight = NSNotFound;
```
if setNeedsLayout isn't needed, are these NSNotFound settings needed at all? if so, why?

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -325,6 +319,30 @@ - (void)setDrawerState:(MDCBottomDrawerState)drawerState {
+
+- (void)setInitialDrawerFactor:(CGFloat)initialDrawerFactor {
+  if ([self shouldPresentFullScreen]) {
+    _initialDrawerFactor = 1;
```
this if conditional is redundant, please remove it. You are also overriding it in the line after.

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -339,6 +357,7 @@ - (void)updateDrawerState:(CGFloat)transitionPercentage {
 - (void)viewDidLoad {
   [super viewDidLoad];
 
+  self.initialDrawerFactor = [self shouldPresentFullScreen] ? 1 : (CGFloat)0.5;
```
this isn't needed i think. The getter makes sure it's 1 if its full screen, and 0.5 is the init value.

yarneo commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,37 @@ - (void)testBottomDrawerTopInset {
+  // Given
+  self.fakeBottomDrawer.trackingScrollView = nil;
+  self.fakeBottomDrawer.originalPresentingViewController.view.bounds = CGRectMake(0, 0, 200, 500);
+  UIViewController *fake = [[UIViewController alloc] init];
```
there is already a fake view controller in the set up, do you need to create this?

yarneo commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,37 @@ - (void)testBottomDrawerTopInset {
+- (void)testExpandToFullScreenAnimationWithSmallScreenContent {
+  // Given
+  self.fakeBottomDrawer.originalPresentingViewController.view.bounds = CGRectMake(0, 0, 200, 500);
+  UIViewController *fakeContentViewController = [[UIViewController alloc] init];
```
there is already a fake view controller in the set up, do you need to create this?

yarneo commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,37 @@ - (void)testBottomDrawerTopInset {
+  [self.fakeBottomDrawer expandToFullScreenWithDuration:0.0];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.fakeBottomDrawer.scrollView.contentOffset.y, 250, 0.001);
```
0.001 needs a CGFloat cast

yarneo commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,37 @@ - (void)testBottomDrawerTopInset {
+  // When
+  [self.fakeBottomDrawer.view setNeedsLayout];
+  [self.fakeBottomDrawer.view layoutIfNeeded];
+  [self.fakeBottomDrawer expandToFullScreenWithDuration:0.0];
```
this can be 0

yarneo commented on components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:
```diff
@@ -544,4 +544,37 @@ - (void)testBottomDrawerTopInset {
+  [self.fakeBottomDrawer expandToFullScreenWithDuration:0.0];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.fakeBottomDrawer.scrollView.contentOffset.y, 50, 0.001);
```
0.001 needs a CGFloat cast

yarneo commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.m:
```diff
@@ -205,4 +205,14 @@ - (void)contentDrawerTopInset:(CGFloat)transitionToTop {
+        (MDCBottomDrawerPresentationController *)self.presentationController;
+    [bottomDrawerPresentationController expandToFullScreenWithDuration:duration];
+  }
+  completion(YES);
```
how are you assuring the completion block is only called after the animation is complete? the animation is done asynchronously, and this will be called immediately after the invocation. Please pass the completion block further in and put it where the completion of the animation block is.

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,46 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+  if (@available(iOS 11.0, *)) {
+    topSafeArea = self.view.safeAreaInsets.top;
+  }
+  CGFloat totalHeight = self.headerViewController.preferredContentSize.height +
```
can you refrain from using the preferredContentSize of the contentViewController? Is it the same as the cached one or has it changed at this moment? have a look at `_contentVCPreferredContentSizeHeightCached`

codeman7 committed:
Clang

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,46 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+      self.contentViewController.preferredContentSize.height;
+  CGFloat precentageOfFullScreen = totalHeight / CGRectGetHeight(self.presentingViewBounds);
+  if (CGRectGetHeight(self.presentingViewBounds) > totalHeight) {
+    CGFloat maxViewHeight = CGRectGetHeight(self.presentingViewBounds) - totalHeight;
```
can you move up properties that are mutual to both the "if" and "else" blocks, i see some similarity.

yarneo commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -447,6 +466,46 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)expandToFullScreenWithDuration:(NSTimeInterval)duration {
```
is the below calculation really needed or do you think it is similar to other calculations we have already made? just making sure we aren't re-calculating twice

jverkoey committed:
Extract clang-format-ci to a separate command. (#5800)

This aligns the clang-format-ci job with how we expect other teams to use it.

Notably, this change introduces a new kokoro job for [clang-format-ci](https://github.com/material-foundation/clang-format-ci) that runs the `.clang-format-ci` script. The kokoro job that runs this script is not configured in this repository - see [go/mdc-kokoro](http://go/mdc-kokoro) for the internal job configuration. This change also turns down the .kokoro clang-format job, so `.kokoro -d clang-format` has been removed.

yarneo committed:
[Release Docs] Updated release docs to be better defined. (#5775)

This PR is an effort to make the release process better defined and more clear about the steps.

yarneo committed:
[Cards] Have updates to MDCCardCollectionCell setState properly respect the animated flag. (#5780)

Previously the highlighted code path did called the animated variants of startTouchBegan / startTouchEnd, irrespective of the animated flag.

This is taken from cl/220520595

codeman7 committed:
Address feedback

codeman7 committed:
Custom height

Colemalban committed:
[ActionSheet]Action sheet safe area fix (#5769)

The MDCActionSheetController currently uses safeAreaInsets to work with the safeArea and alignment. As seen in docs relating to this 
(https://developer.apple.com/documentation/uikit/uiview/2891103-safeareainsets?language=objc)
if the view is not on the screen or in a view hierarchy yet, the values will be zero. I noticed while using  a new iPad pro that the safeAreaInsets were all 0 and the initial layout of the action sheet would cut into the home bar at the bottom. Changed this to use adjustedContentInset (a scroll view property) (https://developer.apple.com/documentation/uikit/uiscrollview/2902259-adjustedcontentinset?language=objc) which fixes the issue.

closes #5750

galiak11 committed:
[Dialogs] renmaing typical use example (#5807)

Part 1 of cleaning up Dialog examples - renaming typical use example.

Part of #5318

codeman7 committed:
Working unless small then large

codeman7 committed:
Refactor for tests

codeman7 committed:
Merge branch 'nav-drawer-present' of https://github.com/codeman7/material-components-ios into nav-drawer-present

codeman7 committed:
Add test / refactor old test

codeman7 committed:
Fix completion / fix test

codeman7 committed:
clang

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:
```diff
@@ -101,4 +101,14 @@
+ */
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^__nullable)(BOOL finished))completion;
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^__nullable)(BOOL finished))completion;
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^__nullable)(BOOL finished))completion;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.m:
```diff
@@ -214,4 +214,16 @@ - (void)setContentOffsetY:(CGFloat)contentOffsetY animated:(BOOL)animated {
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^)(BOOL))completion {
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^)(BOOL))completion {
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^)(BOOL))completion {
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerViewController.m -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.m:
```diff
@@ -214,4 +214,16 @@ - (void)setContentOffsetY:(CGFloat)contentOffsetY animated:(BOOL)animated {
+        (MDCBottomDrawerPresentationController *)self.presentationController;
+    [bottomDrawerPresentationController animateToPreferredContentHeight:preferredContentHeight
+                                                          withDuration:duration
+                                                            completion:completion];
```
clang-format suggested the following change:


```
-                                                          withDuration:duration
-                                                            completion:completion];
+                                                           withDuration:duration
+                                                             completion:completion];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerViewController.m -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerViewController.h:
```diff
@@ -100,6 +100,16 @@
+ */
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^__nullable)(BOOL finished))completion;
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^__nullable)(BOOL finished))completion;
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^__nullable)(BOOL finished))completion;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerViewController.h -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -481,6 +511,80 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^ _Nullable)(BOOL))completion {
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^ _Nullable)(BOOL))completion {
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -481,6 +511,80 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+  CGFloat totalHeight = [self totalHeightWithAddedContentHeight:preferredContentHeight];
+  BOOL setContentHeight = NO;
+  CGSize newPreferredContentSize = CGSizeMake(CGRectGetWidth(self.view.bounds),
+                                              preferredContentHeight);
```
clang-format suggested the following change:


```
-  CGSize newPreferredContentSize = CGSizeMake(CGRectGetWidth(self.view.bounds),
-                                              preferredContentHeight);
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -481,6 +511,80 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+    if (completion) {
+      completion(YES);
+    }
+  }];
```
clang-format suggested the following change:


```
-  [UIView animateWithDuration:duration animations:^{
-    [self.scrollView setContentOffset:contentOffset];
-  } completion:^(BOOL finished) {
-    if (CGRectGetHeight(self.presentingViewBounds) <= totalHeight) {
-      [self.scrollView setContentOffset:CGPointZero];
-    }
-    if (setContentHeight) {
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen
-                          preferredContentSize:newPreferredContentSize];
-    } else {
-      [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
-    }
-    if (completion) {
-      completion(YES);
-    }
-  }];
+  [UIView animateWithDuration:duration
+      animations:^{
+        [self.scrollView setContentOffset:contentOffset];
+      }
+      completion:^(BOOL finished) {
+        if (CGRectGetHeight(self.presentingViewBounds) <= totalHeight) {
+          [self.scrollView setContentOffset:CGPointZero];
+        }
+        if (setContentHeight) {
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen
+                              preferredContentSize:newPreferredContentSize];
+        } else {
+          [self resetLayoutWithInitialDrawerFactor:precentageOfFullScreen];
+        }
+        if (completion) {
+          completion(YES);
+        }
+      }];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m:
```diff
@@ -481,6 +511,80 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
+}
+
+- (void)resetLayoutWithInitialDrawerFactor:(CGFloat)initialDrawerFactor
+                           preferredContentSize:(CGSize)preferredContentSize {
```
clang-format suggested the following change:


```
-                           preferredContentSize:(CGSize)preferredContentSize {
+                      preferredContentSize:(CGSize)preferredContentSize {
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m -->

material-automation commented on components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h:
```diff
@@ -121,4 +121,14 @@
+ */
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^__nullable)(BOOL finished))completion;
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^__nullable)(BOOL finished))completion;
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^__nullable)(BOOL finished))completion;
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m:
```diff
@@ -227,6 +227,15 @@ - (BOOL)contentReachesFullscreen {
   return self.bottomDrawerContainerViewController.contentReachesFullscreen;
 }
 
+
```
clang-format suggested the following change:


```
-
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m:
```diff
@@ -227,6 +227,15 @@ - (BOOL)contentReachesFullscreen {
+
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                          withDuration:(NSTimeInterval)duration
+                            completion:(void (^ _Nullable)(BOOL))completion {
```
clang-format suggested the following change:


```
-                          withDuration:(NSTimeInterval)duration
-                            completion:(void (^ _Nullable)(BOOL))completion {
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m -->

material-automation commented on components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m:
```diff
@@ -227,6 +227,15 @@ - (BOOL)contentReachesFullscreen {
+                            completion:(void (^ _Nullable)(BOOL))completion {
+  [self.bottomDrawerContainerViewController animateToPreferredContentHeight:preferredContentHeight
+                                                              withDuration:duration
+                                                                completion:completion];
```
clang-format suggested the following change:


```
-                                                              withDuration:duration
-                                                                completion:completion];
+                                                               withDuration:duration
+                                                                 completion:completion];
```

<!-- identifier: clang-format -->
<!-- File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m -->
